snapcraft/plugins/kernel.py

```diff
@@ -355,11 +379,78 @@ def _copy_dtbs(self):
             for f in found_dtbs:
                 os.link(f, os.path.join(dtb_dir, os.path.basename(f)))
 
+    def _do_parse_config(self, config_path):
+        # tokenize .config and store options in builtin[] or modules[]
+        with open(config_path) as f:
+            for line in f:
+                tok = line.strip().split('=')
+                items = len(tok)
+                if items == 2:
+                    opt = tok[0].upper()
+                    val = tok[1].upper()
+                    if val == 'Y':
+                        builtin.append(opt)
+                    elif val == 'M':
+                        modules.append(opt)
+
+    def _do_check_config(self):
+        # check the resulting .config has all the necessary options
+        msg = 'Your kernel config is missing some feature that ubuntu core \
+recommend / requires,\nand while we won\'t prevent you from building this\
+kernel snap,\n we suggest you take a look at these:'
```

Comment:
This would look a little prettier like this:
```
msg = ("Your kernel config is missing some features that ubuntu core "
            "recommends or requires,\n"
            "and while we won't prevent you from building this "
            "kernel snap,\n "
            "we suggest you take a look at these:")
```

Note that you also have to correct featureS and recommendS.